test/ValidateID.cpp

```diff
@@ -1694,6 +1694,76 @@ TEST_F(ValidateID, OpPtrAccessChainGood) {
   CHECK_KERNEL(spirv, SPV_SUCCESS, 64);
 }
 
+TEST_F(ValidateID, OpLoadBitcastPointerGood) {
+  const char* spirv = R"(
+%2  = OpTypeVoid
+%3  = OpTypeInt 32 1
+%4  = OpTypeFloat 32
+%5  = OpTypePointer UniformConstant %3
+%6  = OpTypePointer UniformConstant %4
+%7  = OpVariable %5 UniformConstant
+%8  = OpTypeFunction %2
+%9  = OpFunction %2 None %8
+%10 = OpLabel
+%11 = OpBitcast %6 %7
+%12 = OpLoad %4 %11
+      OpReturn
+      OpFunctionEnd)";
+  CHECK_KERNEL(spirv, SPV_SUCCESS, 64);
+}
+TEST_F(ValidateID, OpLoadBitcastNonPointerBad) {
+  const char* spirv = R"(
+%2  = OpTypeVoid
+%3  = OpTypeInt 32 1
+%4  = OpTypeFloat 32
+%5  = OpTypePointer UniformConstant %3
+%6  = OpTypeFunction %2
+%7  = OpVariable %5 UniformConstant
+%8  = OpFunction %2 None %6
+%9  = OpLabel
+%10 = OpLoad %3 %7
+%11 = OpBitcast %4 %10
+%12 = OpLoad %3 %11
```

Comment:
Nope, OpLoad/OpStore first check if the id came from an opcode that can/does return a pointer, then they check if id actually is a pointer, then they check if the result type matches. I had originally hoped that the first check is redundant, but it isn't in logical addressing mode. It _is_ redundant in physical addressing mode however, because it doesn't have pointer opcode restrictions and the type check will make sure that it's actually using a pointer. So a `spvMayReturnPointer` is only necessary for the logical addressing model, which is way I'd rather have a "logical" in that name or always return true when using it with a `SpvAddressingModelPhysical*` argument.

And yes, for the time being it's better to do this check in OpLoad/OpStore (and actually all other pointer taking opcodes ...) rather checking it in all opcodes that can return a pointer (which includes the ones that aren't allowed in logical mode). Until validate_id.cpp is completely done this would probably break too much stuff.